src/main/java/uk/gov/justice/digital/delius/jpa/filters/ContactFilter.java

```diff
@@ -100,6 +113,20 @@ public Predicate toPredicate(Root<Contact> root, CriteriaQuery<?> query, Criteri
         return cb.and(predicates.toArray(new Predicate[0]));
     }
 
+    private void nonRarActivityOnlyFilter(Root<Contact> root, CriteriaBuilder cb, ImmutableList.Builder<Predicate> predicateBuilder) {
+        predicateBuilder.add(cb.equal(root.get("rarActivity"), "N"));
+        predicateBuilder.add(cb.or(cb.isNull(root.get("requirement")),
+            cb.notEqual(root.get("requirement").get("requirementTypeMainCategory")
+                .get("code"), REHABILITATION_ACTIVITY_REQUIREMENT_CODE)));
+    }
+
```

Comment:
Are you sure this "not" logic is correct?
Assuming you are defining a rar activity as both 

rarActivity = Y AND requirement = RAR

then would the not of this be :
rarActivity = N OR  requirement != RAR

BTW I don't know the answer to this since I don't understand the data but worth checking

Reply:
As above - Updated the filter